docs/pallas/tpu/details.rst

```diff
@@ -196,18 +196,45 @@ for those arguments. But, the ``BlockSpec``\s for all subsequent arguments will
 receive not only the grid indices, but also the SMEM references to the leading
 operands.
 
-.. note::
-  We are working on implementing examples for this feature. Stay tuned!
+See (see :ref:`pallas_scalar_prefetch_guide`) for examples on using this
+feature.
+
+Array Layouts
+^^^^^^^^^^^^^
+
+When a vector value is loaded from VMEM into registers (e.g. `x = x_ref[...]`),
+Pallas will tile the last two dimensions of the array into the TPU registers,
+which are typically of size 8x128 for 32-bit values (as of TPU v6).
+
+Here is a graphical example of how a 12x320 array can be tiled using 6 8x128
+tiles:
+
+.. image:: ../../_static/pallas/vector_layout_example.svg
+
+Tiled layouts have several import ramifications for kernel writers:
+
+* The last two axes of an array are treated differently than other
+  axes. For example, reductions, reshapes, and transposes are generally
+  more expensive when involving the last two axes. Some reshapes
+  involving the last two dimensions are not supported and will result in a compiler
+  error, but are "free" and performed at compile time for other dimensions.
+* While sometimes unavoidable, it is generally wasteful to have singleton
+  dimensions in the last two axes, since they will occupy 1 element out of
+  the entire tile dimension. Adding a two 1x1 arrays costs as much as adding
+  two 8x128 arrays, and adding two 8x128x1x1 arrays will be 1024 times as
+  expensive as adding two 8x128 arrays.
```

Comment:
My only suggestion is to mention the word "padding" somewhere here.

Reply:
Done - rewrote this to say that computation is padded up to the tile size.